commonItems/Log.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace commonItems {
+    public enum LogLevel {
+        Error,
+        Warning,
+        Info,
+        Debug,
+        Progress
+    }
+
+    public static class Log {
+        private static bool logFileCreated = false;
+        public static void WriteLine(LogLevel level, string message) {
+            StringBuilder logLine = new();
+            logLine.Append(logLevelStrings[level]);
+            logLine.Append(message);
+            Console.WriteLine(logLine);
+
+            if (!logFileCreated) {
+                System.IO.File.WriteAllText("log.txt", string.Empty);
+                logFileCreated = true;
+            }
+
+            using StreamWriter logFile = File.AppendText("log.txt");
```

Comment:
I'm guessing this closes the file (flushing any write queues) when its garbage collected after falling out of scope. Is there a way to ensure that happens immediately on function exit? Otherwise you might experience a crash but the relevant log information hasn't been output yet.

Reply:
StreamWriter in a `using` block takes cares of closing/disposing
```
            using (StreamWriter logFile = File.AppendText("log.txt"))
            {
                logFile.Write(DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss"));
                logFile.Write(logLine);
                logFile.Write(Environment.NewLine);
            }
```
https://stackoverflow.com/a/11467483/10249243
I think there's no need for manually collecting in that case.

Reply:
Ah, that's pretty cool.